nix/pkgs/haskell/haskell.nix

```diff
@@ -58,9 +57,96 @@ let
       "https://github.com/input-output-hk/hedgehog-extras"."edf6945007177a638fbeb8802397f3a6f4e47c14" = "0wc7qzkc7j4ns2rz562h6qrx2f8xyq7yjcb7zidnj7f6j0pcd0i9";
     };
+    # Configuration settings needed for cabal configure to work when cross compiling
+    # to windows. We can't use `modules` for these as `modules` are only applied
+    # after cabal has been configured.
+    cabalProjectLocal = lib.optionalString pkgs.stdenv.hostPlatform.isWindows ''
+      -- When cross compiling we don't have a `ghc` package
+      package plutus-tx-plugin
+        flags: +use-ghc-stub
+
+      -- The following is needed because Nix is doing something crazy.
+      package byron-spec-ledger
+        tests: False
+
+      package marlowe
+        tests: False
+
+      package plutus-doc
+        tests: False
+
+      package plutus-metatheory
+        tests: False
+
+      package prettyprinter-configurable
+        tests: False
+
+      package small-steps
+        tests: False
+
+      package small-steps-test
+        tests: False
+
+      package byron-spec-chain
+        tests: False
+
+      package cardano-node
+        flags: -systemd
+
+      package cardano-config
+        flags: -systemd
+    '';
     modules = [
-      {
-        reinstallableLibGhc = false;
+      # Allow reinstallation of Win32
+      ({ pkgs, ... }: lib.mkIf pkgs.stdenv.hostPlatform.isWindows {
+        nonReinstallablePkgs =
+          [
+            "rts"
+            "ghc-heap"
+            "ghc-prim"
+            "integer-gmp"
+            "integer-simple"
+            "base"
+            "deepseq"
+            "array"
+            "ghc-boot-th"
+            "pretty"
+            "template-haskell"
+            # ghcjs custom packages
+            "ghcjs-prim"
+            "ghcjs-th"
+            "ghc"
+            "array"
+            "binary"
+            "bytestring"
+            "containers"
+            "filepath"
+            "ghc-compact"
+            "ghc-prim"
+            # "ghci" "haskeline"
+            "hpc"
+            "mtl"
+            "parsec"
+            "text"
+            "transformers"
+            "xhtml"
+            # "stm" "terminfo"
+          ];
+        packages.Win32.components.library.build-tools = lib.mkForce [ ];
+        # TODO fix nix-tools so this is automatic
+        packages.marlowe.components.tests.marlowe-test.buildable = lib.mkForce false;
+        packages.prettyprinter-configurable.components.tests.prettyprinter-configurable-test.buildable = lib.mkForce false;
+      })
+      ({ pkgs, ... }: lib.mkIf (pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform) {
+        # Remove hsc2hs build-tool dependencies (suitable version will be available as part of the ghc derivation)
+        packages.terminal-size.components.library.build-tools = lib.mkForce [ ];
+        packages.network.components.library.build-tools = lib.mkForce [ ];
+        packages.process.components.library.libs = lib.mkForce [ ];
```

Comment:
what's this one?

Reply:
This is gone now too.  These packages include a `build-tool-depends` on `hsc2hs`.  IIRC the plan we generated caused us to try to build hsc2hs, but with the wrong Win32.  Just telling haskell.nix not to try to build it worked.